router/http/src/main/scala/io/buoyant/router/http/ContentLengthFilter.scala

```diff
@@ -0,0 +1,57 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle.Stack.Module0
+import com.twitter.finagle.{Service, ServiceFactory, SimpleFilter, Stack}
+import com.twitter.finagle.http.{Fields, Request, Response, Version}
+import com.twitter.io.Reader
+import com.twitter.util.Future
+
+/**
+ * When Linkerd receives an HTTP/1.0 response that is terminated by closing the connection, the
+ * response will not have a Content-Length header.  This causes Linkerd to interpret the response
+ * as chunk encoded.  To avoid sending chunk encoded responses to HTTP/1.0 clients, we fully buffer
+ * such responses and set the Content-Length header.
+ */
+object ContentLengthFilter {
+
+  object module extends Module0[ServiceFactory[Request, Response]] {
+
+    override def role: Stack.Role = Stack.Role("content-length")
+
+    override def description: String = "Add a content-length header for HTTP/1.0 responses that need one"
+
+    override def make(next: ServiceFactory[Request, Response]): ServiceFactory[Request, Response] =
+      filter.andThen(next)
+  }
+
+  object filter extends SimpleFilter[Request, Response] {
+
+    def hasTransferEncoding(response: Response): Boolean =
+      response.headerMap.get(Fields.TransferEncoding) match {
+        case None => false
+        case Some(te) if te.toLowerCase == "identity" => false
```

Comment:
According to [RFC 7230, section 3.3.1: Transfer-Encoding](https://tools.ietf.org/html/rfc7230#section-3.3.1), Transfer-Encoding header values can be a comma separated string. Is it possible we may run into a value like `chunked, identity`? If so, this may not satisfy the condition. I am not entirely sure if `chunked, identity` is an actual field value that would be generated by an Http/1.0 client.

Reply:
From https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html

> 2.If a Transfer-Encoding header field (section 14.41) is present and has any value other than "identity", then the transfer-length is defined by use of the "chunked" transfer-coding (section 3.6), unless the message is terminated by closing the connection.